src/Extensions.cs

```diff
@@ -76,6 +76,12 @@ public static async Task<IGattService1> GetServiceAsync(this IDevice1 device, st
       return null;
     }
 
+    public static async Task<IGattService1[]> GetServicesAsync(this IDevice1 device)
+    {
+        var services = await BlueZManager.GetProxiesAsync<IGattService1>(BluezConstants.GattServiceInterface, device);
+        return services.ToArray();
+    }
```

Comment:
```suggestion
    public static async Task<IReadOnlyList<GattService1>> GetServicesAsync(this IDevice1 device)
    {
        return await BlueZManager.GetProxiesAsync<IGattService1>(BluezConstants.GattServiceInterface, device);
    }
```

This library favors returning `IReadOnlyList<T>` for collections. `IReadOnlyList<T>` is implemented by both arrays and `List<T>` and is [covariant](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/covariance-contravariance/variance-in-generic-interfaces).

Also `GetProxiesAsync` is already returning a materialized collection; it would be better to avoid copying it to an array.